coil-compose-base/src/main/java/coil/compose/ContentPainterModifier.kt

```diff
@@ -0,0 +1,190 @@
+package coil.compose
+
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.draw.DrawModifier
+import androidx.compose.ui.draw.paint
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.geometry.isSpecified
+import androidx.compose.ui.geometry.isUnspecified
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.drawscope.ContentDrawScope
+import androidx.compose.ui.graphics.drawscope.translate
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.layout.IntrinsicMeasurable
+import androidx.compose.ui.layout.IntrinsicMeasureScope
+import androidx.compose.ui.layout.LayoutModifier
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.MeasureResult
+import androidx.compose.ui.layout.MeasureScope
+import androidx.compose.ui.layout.times
+import androidx.compose.ui.platform.InspectorValueInfo
+import androidx.compose.ui.platform.debugInspectorInfo
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.constrainHeight
+import androidx.compose.ui.unit.constrainWidth
+import kotlin.math.max
+import kotlin.math.roundToInt
+
+/**
+ * A custom [paint] modifier used by [AsyncImageContent] that fills the remaining space
+ * if one dimension of the incoming constraints is fixed.
+ */
+internal data class ContentPainterModifier(
```

Comment:
I created a custom paint modifier since `AsyncImage` computes its size slightly differently than `Image`. We could just have a custom `LayoutModifier` and then set `Modifier.paint`, but I figured it was faster skip all the layout logic in `Modifier.paint` as it isn't used.